dialer.go

```diff
@@ -195,17 +209,20 @@ func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption)
 		endInfo(err)
 		return nil, err
 	}
-	addr, tlsCfg, err := i.ConnectInfo(ctx, cfg.ipType)
+	ci, err := i.ConnectInfo(ctx, cfg.ipType)
 	if err != nil {
 		endInfo(err)
 		return nil, err
 	}
 	endInfo(err)
+	if cErr := verifyIAMAuthNConfig(ci.Version, cfg.refreshCfg.UseIAMAuthN); cErr != nil {
+		return nil, errtype.NewConfigError("instance does not support Auto IAM AuthN", instance)
```

Comment:
This should return the instance version (e.g. "MYSQL_7 does not support Auto IAM AuthN".

Reply:
(AuthN should probably be spelled out here too I guess)